python_modules/libraries/dagster-components/dagster_components/core/component_defs_builder.py

```diff
@@ -67,26 +67,31 @@ def load_components_from_context(context: ComponentLoadContext) -> Sequence[Comp
     raise NotImplementedError(f"Unknown component type {context.decl_node}")
 
 
+def find_local_component_types(component_path: Path) -> list[type[Component]]:
+    """Find all component types defined in a component directory."""
+    component_types = []
+    for py_file in component_path.glob("*.py"):
```

Comment:
Is it possible to reference a component type in a nested directory (e.g. `type: ".foo.bar")? Then we should look recursively.

Reply:
As above, this is factored out of the current logic which finds a matching local component type, so currently you can't - not to say you shouldn't be able to though.